cpp/src/Slice/Parser.h

```diff
@@ -896,8 +893,7 @@ namespace Slice
     class Enumerator final : public virtual Contained
     {
     public:
-        Enumerator(const ContainerPtr&, const std::string&, std::optional<int>);
-        void init() final;
+        Enumerator(const ContainerPtr&, const std::string&, int, bool);
```

Comment:
Please give names to parameters.

Reply:
I also find the previous `std::optional<int>` nicer.

Reply:
> Please give names to parameters.

Alright, I'll do that in a follow up PR.
None of the grammar headers use named parameters. I have no idea why, that's just how it always was.

> I also find the previous std::optional<int> nicer

It doesn't really work here though.
It's probably hard to follow without names, but take:
```
enum Foo { A }
```
Here the enumerator has a name `A`, a value of `0`, and `hasExplicitValue` (the bool) is `false`.
With an `optional`, there's no way to still provide a value in the case when it's '`false`' (ie. empty).

Reply:
ok

Reply:
IMO it's just more straightforward this way. `Enumerator` has always held it's actual value (explicit or implicit), since often, that's all the code gen cares about. A key fact is that `Enumerator` on it's own cannot compute an implicit value for itself.
Only `Enum` has enough information to compute that.

The old logic was `Enum::createEnumerator` would set a special value of `-1` if none was provided.
And the constructor `Enumerator::Enumerator` would call back into `Enum::newEnumerator` to check for this special value and patch it with a real value.

But... there's no benefit to delaying this, instead of just computing the value up front. `Enum` already has all the necessary information. No need to jump back and forth between `Enum` and `Enumerator`, and this avoids storing a special 'replace me' value of `-1` too.